src/GraphBLAS-sharp.Backend/Common/ClArray.fs

```diff
@@ -819,3 +904,44 @@ module ClArray =
 
             processor.Post(Msg.MsgSetArguments(fun () -> kernel.KernelFunc ndRange index array value))
             processor.Post(Msg.CreateRunMsg<_, _> kernel)
+
+    let count (predicate: Expr<'a -> bool>) (clContext: ClContext) workGroupSize =
+
+        let sum =
+            PrefixSum.standardExcludeInPlace clContext workGroupSize
+
+        let getBitmap =
```

Comment:
Wouldn't it be better to convert the `predicate` operation so that it immediately returns `int`? To use `map` here.